pkg/crypto/providers/jwk/jwk.go

```diff
@@ -72,48 +72,71 @@ func (j *JWKEncryptionProvider) GenerateKey(ctx context.Context) (*v2.Encryption
 }
 
-func (j *JWKEncryptionProvider) Encrypt(ctx context.Context, conf *v2.EncryptionConfig, plainText *v2.PlaintextData) (*v2.EncryptedData, error) {
-	jwk, err := unmarshalJWK(conf.GetJwkPublicKeyConfig().GetPubKey())
-	if err != nil {
-		return nil, err
-	}
+func (j *JWKEncryptionProvider) Encrypt(ctx context.Context, configs []*v2.EncryptionConfig, plainText *v2.PlaintextData) ([]*v2.EncryptedData, error) {
+	recipients := make([]age.Recipient, 0, len(configs))
+	encrypted := make([]*v2.EncryptedData, 0, len(configs))
 
-	var ciphertext []byte
-	switch pubKey := jwk.Public().Key.(type) {
-	case ed25519.PublicKey:
-		ciphertext, err = EncryptED25519(pubKey, plainText.Bytes)
+	for _, config := range configs {
+		jwk, err := unmarshalJWK(config.GetJwkPublicKeyConfig().GetPubKey())
 		if err != nil {
 			return nil, err
 		}
-	case *ecdsa.PublicKey:
-		ciphertext, err = EncryptECDSA(pubKey, plainText.Bytes)
-		if err != nil {
-			return nil, err
+		switch pubKey := jwk.Public().Key.(type) {
+		case ed25519.PublicKey:
+			recipient, err := CreateED25519Recipient(pubKey)
+			if err != nil {
+				return nil, err
+			}
+			recipients = append(recipients, recipient)
+		case *rsa.PublicKey:
+			recipient, err := CreateRSARecipient(pubKey)
+			if err != nil {
+				return nil, err
+			}
+			recipients = append(recipients, recipient)
+		case *ecdsa.PublicKey:
+
+			tp, err := thumbprint(jwk)
+			if err != nil {
+				return nil, err
+			}
+
+			ciphertext, err := EncryptECDSA(pubKey, plainText.Bytes)
+			if err != nil {
+				return nil, err
+			}
+			encCipherText := base64.StdEncoding.EncodeToString(ciphertext)
+			encrypted = append(encrypted, &v2.EncryptedData{
+				Provider:       EncryptionProviderJwk,
+				KeyId:          tp,
+				Name:           plainText.Name,
+				Description:    plainText.Description,
+				Schema:         plainText.Schema,
+				EncryptedBytes: []byte(encCipherText),
+			})
+
+		default:
+			return nil, JWKUnsupportedKeyTypeError
 		}
-	case *rsa.PublicKey:
-		ciphertext, err = EncryptRSA(pubKey, plainText.Bytes)
-		if err != nil {
-			return nil, err
-		}
-	default:
-		return nil, JWKUnsupportedKeyTypeError
-	}
-
-	tp, err := thumbprint(jwk)
-	if err != nil {
-		return nil, err
 	}
 
-	encCipherText := base64.StdEncoding.EncodeToString(ciphertext)
-
-	return &v2.EncryptedData{
-		Provider:       EncryptionProviderJwk,
-		KeyId:          tp,
-		Name:           plainText.Name,
-		Description:    plainText.Description,
-		Schema:         plainText.Schema,
-		EncryptedBytes: []byte(encCipherText),
-	}, nil
+	if len(recipients) > 0 {
+		ciphertext, err := ageEncrypt(recipients, plainText.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("jwk: %w", err)
+		}
+		encCipherText := base64.StdEncoding.EncodeToString(ciphertext)
+
+		encrypted = append(encrypted, &v2.EncryptedData{
+			Provider:       EncryptionProviderJwk,
+			KeyId:          "", // MJP what does "keyID" mean for multi-recipient encryption? Multiple ids?
```

Comment:
I'm pretty sure you want to append to encrypted in the loop and set KeyId to config.KeyId. That way cone or whatever can look through the response, find the key ID that it knows, and decrypt that one to provide credentials to the user.

Reply:
Thanks for taking a look Geoff, yeah ecdsa keys aren't supported by the multi-recipient age library we're using (NOTE TO SELF, double check that) so we handle them like we used to. For everything else though, we can actually create just a single EncryptedData package which any key can unlock, this way we don't create (secret * keys) many encrypted blocks, but instead just one that shares keys. My question here is how I should identify said package via KeyID. 
Cone currently doesn't actually filter out based on KeyID (yet, I'm actually working on that next) so we can decide now what form a "multi-key KeyID" should take for our purposes...

A trivial answer could be something like all the KeyIDs we would have, just with some delimiter. meaning we'd have to just parse them out later.

e.g.  "d0cfc64";"79ee7b51b03813b408c";"82f0a1832df2b3ec179f3b";"b47c6b3c9071b724" or what have you.

That's probably my preferred route as opposed to some weird hash of all keys (which limits our ability to ever get a matching hash again) or making EncryptedData store a list of strings which might be more "correct" actually... 🤔